client/analytics/components/report-table/index.js

```diff
@@ -167,29 +167,32 @@ class ReportTable extends Component {
 			endpoint,
 			getHeadersContent,
 			getRowsContent,
+			initiateReportExport,
 			query,
 			searchBy,
 			tableData,
 			title,
-			totalRows,
 		} = this.props;
 		const params = Object.assign( {}, query );
-		const { items } = tableData;
+		const { items, query: reportQuery } = tableData;
+		const { data, totalResults } = items;
 
 		// Delete unnecessary items from filename.
 		delete params.extended_info;
 		if ( params.search ) {
 			delete params[ searchBy ];
 		}
 
-		if ( items.length === totalRows ) {
+		if ( data && data.length === totalResults ) {
 			downloadCSVFile(
 				generateCSVFileName( title, params ),
-				generateCSVDataFromTable( getHeadersContent(), getRowsContent( items.data ) )
+				generateCSVDataFromTable( getHeadersContent(), getRowsContent( data ) )
 			);
+		} else {
+			initiateReportExport( endpoint, title, reportQuery );
 		}
 
-		recordEvent( 'analytics_table_download', { report: endpoint, rows: totalRows } );
+		recordEvent( 'analytics_table_download', { report: endpoint, rows: totalResults } );
```

Comment:
Should we add a parameter to describe what type, ie in-browser or email?

Reply:
Added in https://github.com/woocommerce/woocommerce-admin/pull/2951/commits/094f116d3fc4e01ee3d9dfa78c55f1a2fa1dc039